pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,500 @@
+        add_edge
+        add_edges_from
+        intra_slice
+        inter_slice
+        add_cpds
+        get_cpds
+        compute_initial_state
+        unroll
+        remove_timeslice
+        """
+        super().__init__()
+        self.inter_edges = []
+        self.intra_edges = []
+        self.dbn_cpds = OrderedDict()
+        self.list_of_nodes = []
+        if ebunch:
+            self.add_nodes_from(ebunch)
+        self.timestate = 1
+        self.computed = False
+
+    def add_nodes_from(self, nodes):
+        """
+        adding nodes in the static Bayesian Network.
+        
+        Parameters:
+        ----------
+        nodes: nodes
+               Nodes can be any hashable python object.
+
+        EXAMPLE
+        --------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        """
+        for node in nodes:
+            self.list_of_nodes.extend(node)
+
+    def add_edge(self, start, end, **kwargs):
+        """
+        Add an edge between two nodes.
+
+        The nodes will be automatically added if they are
+        not already in the initial static Bayesian Network.
+
+        Parameters
+        ----------
+        start, end: The start, end nodes should contain the
+                    (node_name, time_slice)
+                    Here, node_name can be a hashable python object
+                    while the time_slice is an integer value, which
+                    denotes the index of the time_slice that the node
+                    belongs to.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['d', 'i'])
+        >>> dbn.add_edge((('d',0), ('i',0)))
+        """
+        edge = (start, end)
+        if not any(map(lambda x: isinstance(x, tuple), iter(edge))):
+            raise AttributeError('The edge is not of a standard form')
+
+        if not any(map(lambda x: isinstance(x[1], int), iter(edge))):
+            raise AttributeError('The edge does not contain the timeslice in integer form')
+
+        if not any(map(lambda x: x[0] in self.list_of_nodes, iter(edge))):
+            new_nodes = list(filter(lambda x: x[0] not in self.list_of_nodes, iter(edge)))
+            self.add_nodes_from(set(map(lambda x: x[0], iter(new_nodes))))
+
+        if edge[0][1] == edge[1][1]:
+            if edge[0][1] in (0, 1):
+                new_edge = map(lambda x: (x[0], 1 - edge[0][1]), edge)
+                super().add_edge(*edge_type(*new_edge))
+                self.intra_edges.extend([edge_type(*edge), edge_type(*map(lambda x: (x[0], 1 - edge[0][1]), edge))])
+            else:
+                self.intra_edges.extend([edge_type(*edge)])
+        else:
+            self.inter_edges.append(edge_type(*edge))
+
+        super().add_edge(*edge_type(*edge))
+
+    def add_edges_from(self, ebunch, **kwargs):
```

Comment:
You don't need to define ```add_edges_from``` method again. It would directly inherit this method from the parent class, where it does the same thing. So in the child class only define the ```add_edge``` method.